spec.md

```diff
@@ -0,0 +1,238 @@
+
+### Defining Layer Types
+
+Artifact layers are expected to be unique, per artifact author. A layer that represents an OCI Image is not expected to be shared with a Helm Chart. To differentiate layers, artifact authors SHOULD create unique artifact layer types.
+
+Artifact layer types follow the same convention as manifests, utilizing the `layer.mediaType` with the following format:  
+`application/vnd.`[org|company]`.`[layerType]`.`[layerSubType]`.layer.`[version]`+`[fileFormat]
+
+### Example Layer Types
+
+|Artifact Type|mediaType|
+|-|-|
+|OCI Image|`application/vnd.oci.image.layer.v1+tar`|
+|Docker Image|`application/vnd.docker.image.rootfs.diff.tar.gzip`|
+|Helm Chart|`application/vnd.cncf.helm.chart.layer.v1+tar`|
+|Singularity SIF Layer|`application/vnd.sylabs.sif.layer.v1+tar`|
+
+### One or More Layer Types
+
+> NOTE: Should this section be in the `spec.md`, or in an `artifact-authors.md` doc?
+
+The number of layer types is up to the artifact author. Some things to consider when designing the layer format:
+
+- **Continuos Builds:** will developers automate building and pushing the artifact to a registry? For each rebuild, what will change? Consider the docker layering model, where users benefit from base layers that don't change often. If only a small piece of content changes, can you separate that content into a unique layer?
+- **Layer reuse**: when layers are pushed to a registry, layers can be de-duped. If your artifact type may benefit from shared layers across many artifact instances, consider splitting up the layers to those that change often, and those that don't.
+- **Layer sizes**: does your artifact have large content that may benefit from being split up in to smaller elements that can be concurrently uploaded and/or downloaded to a registry?
+- **Content and Config**: While a general best practice for container images says the image in the registry should not contain environmental configuration, you may be choosing to push environmental configuration to a registry as it's own artifact. If the same artifact is pushed to the same registry multiple times, only unique by some configuration, consider splitting up the configuration into its own layer, to benefit from layer reuse.
+
+### Shared Layer Example
+
+Consider an Azure Resource Manager (ARM) or AWS Cloud Formation template stored in a registry. You may have the base template stored for shared team use. As teams deploy the template, consider storing the parameters file as a separate blobs. When pushed to a registry, most registry operators will de-dupe the base template, only storing the unique parameters blob for each artifact.
+
+## Optional: Defining Config Schema
+
+While the value of `manifest.config.mediaType` is used to determine the artifact type, the persistance of a `config.json` file is OPTIONAL. Artifacts can push a null reference for `config.json` persistance.
```

Comment:
Reaching inside an image-spec type and relaxing [one of their conditions][1] on one of their types is pretty aggressive.  If you think the property should be optional, either talk them into relaxing their constraint (might have backwards-compat ramifications) or define a new media type that you control.

[1]: https://github.com/opencontainers/image-spec/blame/v1.0.1/manifest.md#L30

Reply:
Can you clarify what you’re referring to?
What i’m saying here is each artifact author can define a config schema. The oci.imgae has a schema that is defined by the oci image spec. The helm or cnab spec may define their config schema as it relates to them. 
By having a manifest.config.mediaType, registry operators and artifact tools know what config schema to expect.

Reply:
`manifest.config.digest` is a required value of the image-spec. For registry operators and oci-image tools processing and validating oci image artifacts, they should continue. 
What we're saying is oci-image is just one of many artifact types stored in a registry.
Other artifacts can push null for the config object. 
[Here's an ORAS example](https://github.com/deislabs/oras#pushing-artifacts-with-single-files): 
```sh
oras push localhost:5000/hello-artifact:v1 \
--manifest-config /dev/null:application/vnd.acme.rocket.config.v1+json \
./artifact.txt
```
Registry operators may choose to opt-in to parsing specific artifact config objects. However, it's not required.

Reply:
For OCI Image, the value of `manifest.config` must be populated. 
For other artifacts, this is up to the artifact author. 
What I think we're discussing is whether an instance of the distribution spec MUST have a `manifest.config` object. 
This is a bit vague, or was assumed as distribution and image-spec were presumed as one. This spec explicitly states a config value is optional. 
FWIW, the current docker/distribution reference implementation doesn't require a config object.
I'll close this one as I think this helps with the differentiation.